### DISCUSSION TITLE
finding current request path

### DISCUSSION THREAD

**Discussion by kamer2004** [mojolicious/mojo#1872] [Q&A]
Hi, what have I please to write in this sample to print, through $here, '/some/location' ?

get '/some/location' => sub ($c) {
...
$here = ...?
print $here ."\n";
...
} => /some/location';

thanks

**Comment by christopherraa** [comment]
Hi there,

You should have a look at [Mojolicious::Controller->req](https://mojolicio.us/perldoc/Mojolicious/Controller#req) that gives you a [Mojo::Message::Request](https://docs.mojolicious.org/Mojo/Message/Request) object, holding among other things the URL. It would probably be usefulfor you to familiarize yourself with the [documentation](https://docs.mojolicious.org/), and also [the guides](https://docs.mojolicious.org/#GUIDES). Lots of great information there.

**Reply by s1037989** [reply]
The code:
```
$ cat myapp.pl 
#!/usr/bin/env perl
use Mojolicious::Lite -signatures;

get '/some/location' => sub ($c) {
  my $here = $c->req->url;                # <--  The Request object @christopherraa referred to
  warn $here->to_abs . "\n";
  $c->render(text => $here);
};

app->start;
```

The demo:
```
$ perl myapp.pl get -m production /some/location
http://127.0.0.1:45253/some/location
/some/location
```
